runtime/legion/legion_ops.h

```diff
@@ -670,6 +688,22 @@ namespace Legion {
       RtUserEvent predicate_waiter; // used only when needed
     };
 
+    /**
+     * \class MemoizableOp
+     * A memoizable operation is an abstract class
+     * that serves as the basis for operation whose
+     * physical analysis can be memoized.  Memoizable
+     * operations go through an extra step in the mapper
+     * to determine whether to memoize their physical analysis.
+     */
+    template<typename OP>
+    class MemoizableOp : public OP
+    {
+    public:
+      MemoizableOp(Runtime *rt);
+      virtual void execute_dependence_analysis(void);
```

Comment:
Is this really all we need for a memoizable operation? I was expecting there to be more things. :)

Reply:
For example, I feel like this could capture the 'invoke_memoize' method for most of the operation classes.